lang/rust/src/decimal.rs

```diff
@@ -45,30 +67,271 @@ impl Decimal {
         let sign_byte = 0xFF * u8::from(self.value.sign() == Sign::Minus);
         let mut decimal_bytes = vec![sign_byte; len];
         let raw_bytes = self.value.to_signed_bytes_be();
-        let num_raw_bytes = raw_bytes.len();
-        let start_byte_index = len.checked_sub(num_raw_bytes).ok_or(Error::SignExtend {
+        let raw_bytes_len = raw_bytes.len();
+        let start_byte_index = len.checked_sub(raw_bytes_len).ok_or(Error::SignExtend {
             requested: len,
-            needed: num_raw_bytes,
+            needed: raw_bytes_len,
         })?;
         decimal_bytes[start_byte_index..].copy_from_slice(&raw_bytes);
         Ok(decimal_bytes)
     }
+
+    pub fn from_f64(decimal: f64) -> AvroResult<Self> {
```

Comment:
I'm not convinced this method should be provided. Looking at the `rust_decimal` library it seems to be rather complicated to implement correctly: `https://docs.rs/rust_decimal/1.21.0/src/rust_decimal/decimal.rs.html#1861`

I think `from_bytes(bytes, scale, precision)` makes sense. And something based on integers would make sense:

```rust
from_i128(mantissa: i128, scale: u32, precision: u32)
```
If users want to convert from `f64` they can go via `rust_decimal`.